### PR TITLE
views: rest files API implementation

### PR DESCRIPTION
**Pull request by hachreak**
* Implements files REST API. (addresses zenodo/zenodo#522)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

**Comment by hachreak** [comment]
note: waiting for https://github.com/inveniosoftware/invenio-search/pull/66 and all PR connected.

**Comment by hachreak** [comment]
@lnielsen if you have free time, you can start to review of my PR. thanks

File: invenio_deposit/views/rest.py [hunk 1]
```diff
@@ -96,3 +127,95 @@ def post(self, pid, record, action):
...
+    view_name = '{0}_files'
+
+    def __init__(self, serializers, pid_type, *args, **kwargs):
+        """Constructor."""
+        super(DepositFilesResource, self).__init__(
+            serializers,
+            *args,
+            **kwargs
+        )
+        self.resolver = Resolver(
```
Comment (jirikuncar) [inline invenio_deposit/views/rest.py]: see https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/views.py#L158-L160

**Comment by hachreak** [comment]
@lnielsen I implemented also the sort!
p.s. probably we need to merge my PR with the PR made by @JavierDelgadoFernandez  because he's managing with `deposit['files']` also.

**Comment by jirikuncar** [comment]
closed via #34

**Comment by hachreak** [comment]
@jirikuncar  did you mean #36?

**Pull request closed without merging**